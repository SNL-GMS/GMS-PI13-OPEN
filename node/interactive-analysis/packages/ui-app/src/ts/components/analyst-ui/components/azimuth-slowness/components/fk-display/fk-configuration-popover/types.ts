import { FkTypes } from '@gms/common-graphql';
import { CheckboxSearchListTypes } from '@gms/ui-core-components';
import { FkConfigurationWithUnits, FkUnits } from '~analyst-ui/components/azimuth-slowness/types';

export enum FkConfigurationPopoverPanel {
  DEFAULT,
  ADVANCED
}

export interface FkConfigurationPopoverState {
  openPanel: FkConfigurationPopoverPanel;
  fkUnits: FkUnits;
  normalizeWaveforms: boolean;
  mediumVelocity: number;
  maximumSlowness: number;
  numberOfPoints: number;
  channelCheckboxes: CheckboxSearchListTypes.CheckboxItem[];
  useVerticalChannelOffsets: boolean;
}
export interface FkConfigurationPopoverProps extends FkTypes.FkConfiguration {
  fkUnitDisplayed: FkUnits;
  applyFkConfiguration(configuration: FkConfigurationWithUnits);
  close();
}
