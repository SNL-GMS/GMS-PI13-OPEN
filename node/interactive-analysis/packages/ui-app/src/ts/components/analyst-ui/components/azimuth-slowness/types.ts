import {
  CommonTypes,
  ConfigurationTypes,
  EventTypes,
  FkTypes,
  ProcessingStationTypes,
  SignalDetectionTypes,
  WaveformTypes
} from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { Client } from '@gms/ui-apollo';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { ChildProps, MutationFunction } from 'react-apollo';
import { FilterType, FkThumbnailSize } from './components/fk-thumbnail-list/fk-thumbnails-controls';
import { LeadLagPairs } from './constants';

export interface FkParams {
  windowParams: FkTypes.WindowParameters;
  frequencyPair: FkTypes.FrequencyBand;
}

export interface LeadLagPairAndString {
  leadLagPairs: LeadLagPairs;
  windowParams: FkTypes.WindowParameters;
}

export enum FkUnits {
  FSTAT = 'FSTAT',
  POWER = 'POWER'
}

/**
 * Used to return a super set of the fk configuration from the fk config popover
 */
export interface FkConfigurationWithUnits extends FkTypes.FkConfiguration {
  fkUnitToDisplay: FkUnits;
}

/**
 * Azimuth Slowness Redux Props
 */
export interface AzimuthSlownessReduxProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  client: Client;
  currentStageInterval: AnalystWorkspaceTypes.StageInterval;
  currentTimeInterval: CommonTypes.TimeRange;
  selectedSdIds: string[];
  openEventId: string;
  sdIdsToShowFk: string[];
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  location: AnalystWorkspaceTypes.LocationSolutionState;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  setSelectedSdIds(ids: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setSdIdsToShowFk(signalDetectionIds: string[]): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}

export interface SubscriptionAction {
  (
    list: SignalDetectionTypes.SignalDetection[],
    index: number,
    prev: SignalDetectionTypes.SignalDetection[],
    currentIteree: SignalDetectionTypes.SignalDetection
  ): void;
}

/**
 * Azimuth Slowness State
 */
export interface AzimuthSlownessState {
  fkThumbnailSizePx: FkThumbnailSize;
  fkThumbnailColumnSizePx: number;
  filterType: FilterType;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  userInputFkFrequency: FkTypes.FrequencyBand;
  numberOfOutstandingComputeFkMutations: number;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  fkInnerContainerWidthPx: number;
  fkFrequencyThumbnails: Immutable.Map<string, FkTypes.FkFrequencyThumbnail[]>;
}

/**
 * Mutations used by the Az Slow display
 */
export interface AzimuthSlownessMutations {
  computeFks: MutationFunction<{}>;
  computeFkFrequencyThumbnails: MutationFunction<{}>;
  setWindowLead: MutationFunction<{}>;
  markFksReviewed: MutationFunction<{}>;
}

/**
 * Consolidated props for Azimuth Slowness
 */
export type AzimuthSlownessProps = AzimuthSlownessReduxProps &
  ChildProps<AzimuthSlownessMutations> &
  ProcessingStationTypes.DefaultStationsQueryProps &
  ConfigurationTypes.UIConfigurationQueryProps &
  SignalDetectionTypes.SignalDetectionsByStationQueryProps &
  EventTypes.EventsInTimeRangeQueryProps;

/**
 * State of the az slow panel
 */
export interface AzimuthSlownessPanelState {
  currentMovieSpectrumIndex: number;
}

export interface AzimuthSlownessPanelProps {
  // Data
  defaultStations: ProcessingStationTypes.ProcessingStation[];
  eventsInTimeRange: EventTypes.Event[];
  displayedSignalDetection: SignalDetectionTypes.SignalDetection | undefined;
  openEvent: EventTypes.Event | undefined;
  associatedSignalDetections: SignalDetectionTypes.SignalDetection[];
  signalDetectionsToDraw: SignalDetectionTypes.SignalDetection[];
  signalDetectionsIdToFeaturePredictions: Immutable.Map<string, EventTypes.FeaturePrediction[]>;
  signalDetectionsByStation: SignalDetectionTypes.SignalDetection[];
  featurePredictionsForDisplayedSignalDetection: EventTypes.FeaturePrediction[];
  distances: EventTypes.LocationToStationDistance[];
  // Redux state as props
  selectedSdIds: string[];
  defaultWaveformFilters: WaveformTypes.WaveformFilter[];
  sdIdsToShowFk: string[];
  location: AnalystWorkspaceTypes.LocationSolutionState;
  fkFrequencyThumbnails: FkTypes.FkFrequencyThumbnail[];
  // Azimuth display state as props
  fkThumbnailColumnSizePx: number;
  fkDisplayWidthPx: number;
  fkDisplayHeightPx: number;
  filterType: FilterType;
  fkThumbnailSizePx: FkThumbnailSize;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  numberOfOutstandingComputeFkMutations: number;
  userInputFkFrequency: FkTypes.FrequencyBand;
  fkUnitForDisplayedSignalDetection: FkUnits;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  fkInnerContainerWidthPx: number;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  // Prop functions
  adjustFkInnerContainerWidth(
    fkThumbnailsContainer: HTMLDivElement,
    fkThumbnailsInnerContainer: HTMLDivElement
  ): void;
  markFksForSdIdsAsReviewed(sdIds: string[]): void;
  updateFkThumbnailSize(size: FkThumbnailSize): void;
  updateFkFilter(filterType: FilterType): void;
  setFkThumbnailColumnSizePx(newSizePx: number): void;
  computeFkAndUpdateState(fkInput: FkTypes.FkInput): void;
  changeUserInputFks(
    windowParams: FkTypes.WindowParameters,
    frequencyBand: FkTypes.FrequencyBand
  ): void;
  setFkUnitForSdId(sdId: string, fkUnit: FkUnits): void;

  // Redux functions
  setSelectedSdIds(sdIds: string[]): void;
  setSdIdsToShowFk(sdIds: string[]): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>): void;
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
}
