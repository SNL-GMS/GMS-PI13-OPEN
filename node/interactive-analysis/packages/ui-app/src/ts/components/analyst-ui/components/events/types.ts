import {
  CommonTypes,
  EventTypes,
  ProcessingStationTypes,
  SignalDetectionTypes
} from '@gms/common-graphql';
import GoldenLayout from '@gms/golden-layout';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import { ChildProps, MutationFunction } from 'react-apollo';
import { gmsColors, semanticColors } from '~scss-config/color-preferences';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}
export const eventFilterToColorMap: Map<any, string> = new Map();
eventFilterToColorMap.set(EventFilters.EDGE, gmsColors.gmsBackground);
eventFilterToColorMap.set(EventFilters.COMPLETED, semanticColors.analystComplete);

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction<{}>;
  saveEvent: MutationFunction<{}>;
  saveAllModifiedEvents: MutationFunction<{}>;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analystActivity: AnalystWorkspaceTypes.AnalystActivity;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}
export interface SignalDetectionHypothesisWithStation
  extends SignalDetectionTypes.SignalDetectionHypothesis {
  stationName?: string;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  EventTypes.EventsInTimeRangeQueryProps &
  ProcessingStationTypes.DefaultStationsQueryProps &
  SignalDetectionTypes.SignalDetectionsByStationQueryProps &
  CommonTypes.WorkspaceStateProps;
