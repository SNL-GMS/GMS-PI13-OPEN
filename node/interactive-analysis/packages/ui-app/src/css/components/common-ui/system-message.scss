$critical-selector: '[data-severity*=CRITICAL]';
$warning-selector: '[data-severity*=WARNING]';
$info-selector: '[data-severity*=INFO]';

@mixin createColoredCount($selector, $color) {
  #{$selector} &__count {
    color: lighten($color, 10%);
  }
}

.system-message-display {
  @include flexColumn;
  width: 100%;
  height: 100%;
  .toolbar {
    margin-bottom: 0.5em;
  }
}

.system-message-table {
  @include flexColumn;
  width: 100%;
  height: 100%;
  background-color: $gms-prominent-background;
  .system-cell {
    &#{$critical-selector} {
      background-color: $system-message-critical-color;
    }
    &#{$warning-selector} {
      background-color: $system-message-warning-color;
    }
    &#{$info-selector} {
      background-color: $system-message-info-color;
    }
  }
  &__button {
    &--floating {
      position: absolute;
      left: 50%;
      bottom: 3rem;
      transform: translate3D(-50%, -50%, 0);
    }
    &--desc {
      bottom: auto;
      top: 9rem;
    }
  }
}

.system-message-summary {
  @include centerChildren;
  background: $gms-prominent-background;
  height: 100%;
  padding: 0 1em;
  border-radius: $rounded-rectangle-radius;
  font-size: 1rem;

  @include createColoredCount($critical-selector, $gms-strong-warning);
  @include createColoredCount($warning-selector, $gms-warning);
  @include createColoredCount($info-selector, $gms-good-count);

  &__entry {
    @include centerChildren('vertical');
    cursor: pointer;
    & + & {
      margin-left: 1.5em;
    }
    &--disabled .system-message-summary__count {
      color: $gms-soft;
    }
  }

  &__count {
    font-family: $gms-mono;
    font-size: 1.5rem;
    font-weight: bold;
    margin-left: 0.5ch;
    &--prefix {
      opacity: 0.33;
    }
    &--main {
      opacity: 1;
    }
  }
}

/// AG Grid Overrides
.system-message-table {
  .ag-body-viewport,
  .ag-root-wrapper {
    scroll-behavior: smooth;
  }
  .ag-row .ag-cell {
    background-color: lighten($gms-prominent-background, 5%);
    border-left: 1px solid $gms-prominent-background;
    border-right: 1px solid $gms-prominent-background;
  }
  .ag-header-cell {
    padding: 0 10px;
    background-color: $gms-popover-background;
    .ag-header-cell-text {
      color: $gms-main;
    }
  }
  .ag-header-cell-label,
  .ag-header-cell-label {
    padding-left: 7px;
    font-size: 1rem;
    font-weight: 500;
  }
}
