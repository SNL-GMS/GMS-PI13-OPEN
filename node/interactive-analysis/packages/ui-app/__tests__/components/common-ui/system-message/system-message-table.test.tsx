import * as React from 'react';
import { uuid } from '../../../../../common-util/src/ts/common-util';
import { SystemMessageTable } from '../../../../src/ts/components/common-ui/components/system-message/system-message-table';
import { buildDefaultSeverityFilterMap } from '../../../../src/ts/components/common-ui/components/system-message/toolbar/severity-filters';
import { SystemMessageTableProps } from '../../../../src/ts/components/common-ui/components/system-message/types';
import { systemMessages } from './shared-data';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

// tslint:disable-next-line: deprecation
const lodash = require.requireActual('lodash');
lodash.uniqueId = () => '1';

let idCount = 0;
uuid.asString = jest.fn().mockImplementation(() => ++idCount);

describe('System Message Table', () => {
  it('should be defined', () => {
    expect(SystemMessageTable).toBeDefined();
  });

  const severityFilterMap = buildDefaultSeverityFilterMap();

  const props: SystemMessageTableProps = {
    addSystemMessages: jest.fn(),
    clearAllSystemMessages: jest.fn(),
    clearExpiredSystemMessages: jest.fn(),
    clearSystemMessages: jest.fn(),
    isAutoScrollingEnabled: true,
    setIsAutoScrollingEnabled: jest.fn(),
    systemMessages,
    severityFilterMap
  };

  const table = Enzyme.mount(<SystemMessageTable {...props} />);
  it('should match a snapshot', () => {
    expect(table).toMatchSnapshot();
  });

  const instance = table.instance();

  instance.state = { hasUnseenMessages: true };

  it('should not sort if props change if auto scrolling is enabled and sorted by time', () => {
    instance.forceUpdate = jest.fn();
    instance.table = {
      getTableApi: () => ({
        getSortModel: jest.fn(() => [
          {
            colId: 'time',
            sort: 'asc'
          }
        ]),
        ensureIndexVisible: jest.fn(() => true),
        refreshInfiniteCache: jest.fn()
      })
    };
    instance.sortByTime = jest.fn();
    instance.componentDidUpdate(
      {
        isAutoScrollingEnabled: false
      },
      {
        systemMessages
      }
    );
    clearTimeout(instance.autoScrollTimer);
    expect(instance.sortByTime).not.toHaveBeenCalled();
  });

  it('should sort if props change if auto scrolling is enabled', () => {
    instance.table = {
      getTableApi: () => ({
        getSortModel: jest.fn(() => [
          {
            colId: 'message',
            sort: 'asc'
          }
        ]),
        ensureIndexVisible: jest.fn(() => true),
        refreshInfiniteCache: jest.fn()
      })
    };
    instance.sortByTime = jest.fn();
    instance.componentDidUpdate(
      {
        isAutoScrollingEnabled: false
      },
      {
        systemMessages
      }
    );
    expect(instance.sortByTime).toHaveBeenCalled();
  });

  it('can tell if it has scrolled to the end', () => {
    instance.getScrollContainer = () => ({
      scrollTop: 100, // how far from the top we have scrolled
      scrollHeight: 1000, // simulated height of whole container
      clientHeight: 900 // simulated amount visible
    });
    expect(instance.isScrolledToBottom()).toBeFalsy();
  });

  it('can tell if it has NOT scrolled to the end', () => {
    instance.getScrollContainer = () => ({
      scrollTop: 0, // how far from the top we have scrolled
      scrollHeight: 1000, // simulated height of whole container
      clientHeight: 900 // simulated amount visible
    });
    expect(instance.isScrolledToBottom()).toBeFalsy();
  });

  it('gets the right subcategory values', () => {
    const params = {
      data: {
        id: systemMessages[0].id
      }
    };
    const result = instance.subcategoryValueGetter(params);
    expect(result).toEqual(systemMessages[0].subCategory);
  });

  it('gets the right severity values', () => {
    const params = {
      data: {
        id: systemMessages[0].id
      }
    };
    const result = instance.severityValueGetter(params);
    expect(result).toEqual(systemMessages[0].severity);
  });

  it('gets the right message values', () => {
    const params = {
      data: {
        id: systemMessages[0].id
      }
    };
    const result = instance.messageValueGetter(params);
    expect(result).toEqual(systemMessages[0].message);
  });

  it('can checked if scrolled to start', () => {
    const result = instance.isScrolledToTop();
    expect(result).toBeTruthy();
  });
  it('can handle sort changed', () => {
    instance.onSortChanged();
    expect(instance.props.setIsAutoScrollingEnabled).toHaveBeenCalledTimes(1);
  });

  it('can tell scroll direction down', () => {
    // tslint:disable-next-line: no-magic-numbers
    instance.state.prevScrollPositions = [0, 10];
    expect(instance.getScrollDirection()).toEqual('down');
  });

  it('can tell scroll direction up', () => {
    // tslint:disable-next-line: no-magic-numbers
    instance.state.prevScrollPositions = [Number(instance.SCROLL_UP_THRESHOLD_PX) + 1, 0];
    expect(instance.getScrollDirection()).toEqual('up');
  });

  it('can tell scroll direction unchanged', () => {
    // tslint:disable-next-line: no-magic-numbers
    instance.state.prevScrollPositions = [10, 10];
    expect(instance.getScrollDirection()).toEqual('unchanged');
  });

  it('can use new messages indicator', () => {
    instance.table = {
      getTableApi: () => ({
        getSortModel: jest.fn(() => [
          {
            colId: 'time',
            sort: 'asc'
          }
        ]),
        ensureIndexVisible: jest.fn(() => true)
      })
    };
    instance.onNewMessageIndicatorClick();
    expect(instance.props.setIsAutoScrollingEnabled).toHaveBeenCalledWith(true);
  });

  it('can use on body scroll', () => {
    jest.useFakeTimers();
    instance.props.setIsAutoScrollingEnabled(true);
    instance.getScrollDirection = jest.fn().mockImplementation(() => 'up');
    instance.onBodyScroll();
    jest.runAllTimers();
    expect(instance.getScrollDirection).toHaveBeenCalled();
    expect(instance.props.setIsAutoScrollingEnabled).toHaveBeenLastCalledWith(false);
  });
});
