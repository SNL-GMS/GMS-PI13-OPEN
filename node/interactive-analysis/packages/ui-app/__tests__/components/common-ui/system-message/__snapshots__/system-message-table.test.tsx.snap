// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System Message Table should match a snapshot 1`] = `
<SystemMessageTable
  addSystemMessages={[MockFunction]}
  clearAllSystemMessages={[MockFunction]}
  clearExpiredSystemMessages={[MockFunction]}
  clearSystemMessages={[MockFunction]}
  isAutoScrollingEnabled={true}
  setIsAutoScrollingEnabled={[MockFunction]}
  severityFilterMap={
    Immutable.Map {
      "INFO": true,
      "WARNING": true,
      "CRITICAL": true,
    }
  }
  systemMessages={
    Array [
      Object {
        "category": "SOH",
        "id": "mock1",
        "message": "mock message text",
        "severity": "CRITICAL",
        "subCategory": "STATION",
        "time": 1593030714074,
        "type": "STATION_NEEDS_ATTENTION",
      },
      Object {
        "category": "SOH",
        "id": "mock2",
        "message": "mock message text",
        "severity": "CRITICAL",
        "subCategory": "STATION",
        "time": 1593030714075,
        "type": "STATION_NEEDS_ATTENTION",
      },
    ]
  }
>
  <div
    className="system-message-table"
    data-cy="system-message-table"
    id="system-message-table"
  >
    <Component>
      <InfiniteTable
        alwaysShowVerticalScroll={true}
        columnDefs={
          Array [
            Object {
              "colId": "id",
              "field": "id",
              "headerName": "ID",
              "headerTooltip": "ID",
              "hide": true,
              "width": 10,
            },
            Object {
              "cellRendererFramework": [Function],
              "colId": "time",
              "headerName": "Timestamp",
              "headerTooltip": "Timestamp",
              "sort": "asc",
              "valueFormatter": [Function],
              "valueGetter": [Function],
              "width": 200,
            },
            Object {
              "cellRendererFramework": [Function],
              "colId": "category",
              "comparator": [Function],
              "headerName": "Category",
              "headerTooltip": "Category",
              "valueGetter": [Function],
              "width": 110,
            },
            Object {
              "cellRendererFramework": [Function],
              "colId": "subCategory",
              "comparator": [Function],
              "headerName": "Subcategory",
              "headerTooltip": "Subcategory",
              "valueGetter": [Function],
              "width": 150,
            },
            Object {
              "cellRendererFramework": [Function],
              "colId": "severity",
              "comparator": [Function],
              "headerName": "Severity",
              "headerTooltip": "Severity",
              "valueGetter": [Function],
              "width": 105,
            },
            Object {
              "cellRendererFramework": [Function],
              "colId": "message",
              "comparator": [Function],
              "flex": 1,
              "headerName": "Message",
              "headerTooltip": "Message",
              "minWidth": 200,
              "valueGetter": [Function],
            },
          ]
        }
        datasource={
          Object {
            "destroy": [Function],
            "getRows": [Function],
          }
        }
        debug={false}
        defaultColDef={
          Object {
            "cellRendererFramework": [Function],
            "disableStaticMarkupForCellRendererFramework": true,
            "disableStaticMarkupForHeaderComponentFramework": true,
            "filter": false,
            "headerComponentParams": Object {
              "template": "
  <div class=\\"ag-cell-label-container\\" role=\\"presentation\\">
      <span ref=\\"eMenu\\" class=\\"ag-header-icon ag-header-cell-menu-button\\"></span>
      <div ref=\\"eLabel\\" class=\\"ag-header-cell-label\\" role=\\"presentation\\">
          <span ref=\\"eText\\" class=\\"ag-header-cell-text\\" role=\\"columnheader\\"></span>
          <span ref=\\"eFilter\\" class=\\"ag-header-icon ag-filter-icon\\"></span>
          <!-- <span ref=\\"eSortOrder\\" class=\\"ag-header-icon ag-sort-order\\"></span> -->
          <span ref=\\"eSortAsc\\" class=\\"ag-header-icon ag-sort-ascending-icon\\"></span>
          <span ref=\\"eSortDesc\\" class=\\"ag-header-icon ag-sort-descending-icon\\"></span>
          <span ref=\\"eSortNone\\" class=\\"ag-header-icon ag-sort-none-icon\\"></span>
      </div>
  </div>
",
            },
            "sortable": true,
          }
        }
        headerHeight={44}
        id="system-message-table"
        key="system-message-table"
        onBodyScroll={[Function]}
        onCellClicked={[Function]}
        onCellContextMenu={[Function]}
        onGridReady={[Function]}
        onSortChanged={[Function]}
        overlayNoRowsTemplate="No System Messages"
        pagination={false}
        rowData={
          Array [
            Object {
              "category": "SOH",
              "id": "mock1",
              "message": "mock message text",
              "severity": "CRITICAL",
              "subCategory": "STATION",
              "time": 1593030714074,
              "type": "STATION_NEEDS_ATTENTION",
            },
            Object {
              "category": "SOH",
              "id": "mock2",
              "message": "mock message text",
              "severity": "CRITICAL",
              "subCategory": "STATION",
              "time": 1593030714075,
              "type": "STATION_NEEDS_ATTENTION",
            },
          ]
        }
        rowHeight={40}
        suppressCellSelection={true}
        suppressScrollOnNewData={true}
      >
        <CoreTable
          alwaysShowVerticalScroll={true}
          blockLoadDebounceMillis={60}
          cacheBlockSize={100}
          cacheOverflowSize={10}
          columnDefs={
            Array [
              Object {
                "colId": "id",
                "field": "id",
                "headerName": "ID",
                "headerTooltip": "ID",
                "hide": true,
                "width": 10,
              },
              Object {
                "cellRendererFramework": [Function],
                "colId": "time",
                "headerName": "Timestamp",
                "headerTooltip": "Timestamp",
                "sort": "asc",
                "valueFormatter": [Function],
                "valueGetter": [Function],
                "width": 200,
              },
              Object {
                "cellRendererFramework": [Function],
                "colId": "category",
                "comparator": [Function],
                "headerName": "Category",
                "headerTooltip": "Category",
                "valueGetter": [Function],
                "width": 110,
              },
              Object {
                "cellRendererFramework": [Function],
                "colId": "subCategory",
                "comparator": [Function],
                "headerName": "Subcategory",
                "headerTooltip": "Subcategory",
                "valueGetter": [Function],
                "width": 150,
              },
              Object {
                "cellRendererFramework": [Function],
                "colId": "severity",
                "comparator": [Function],
                "headerName": "Severity",
                "headerTooltip": "Severity",
                "valueGetter": [Function],
                "width": 105,
              },
              Object {
                "cellRendererFramework": [Function],
                "colId": "message",
                "comparator": [Function],
                "flex": 1,
                "headerName": "Message",
                "headerTooltip": "Message",
                "minWidth": 200,
                "valueGetter": [Function],
              },
            ]
          }
          context={
            Object {
              "columnApi": undefined,
              "tableApi": undefined,
            }
          }
          datasource={null}
          debug={false}
          defaultColDef={
            Object {
              "cellRendererFramework": [Function],
              "disableStaticMarkupForCellRendererFramework": true,
              "disableStaticMarkupForHeaderComponentFramework": true,
              "filter": false,
              "headerComponentParams": Object {
                "template": "
  <div class=\\"ag-cell-label-container\\" role=\\"presentation\\">
      <span ref=\\"eMenu\\" class=\\"ag-header-icon ag-header-cell-menu-button\\"></span>
      <div ref=\\"eLabel\\" class=\\"ag-header-cell-label\\" role=\\"presentation\\">
          <span ref=\\"eText\\" class=\\"ag-header-cell-text\\" role=\\"columnheader\\"></span>
          <span ref=\\"eFilter\\" class=\\"ag-header-icon ag-filter-icon\\"></span>
          <!-- <span ref=\\"eSortOrder\\" class=\\"ag-header-icon ag-sort-order\\"></span> -->
          <span ref=\\"eSortAsc\\" class=\\"ag-header-icon ag-sort-ascending-icon\\"></span>
          <span ref=\\"eSortDesc\\" class=\\"ag-header-icon ag-sort-descending-icon\\"></span>
          <span ref=\\"eSortNone\\" class=\\"ag-header-icon ag-sort-none-icon\\"></span>
      </div>
  </div>
",
              },
              "sortable": true,
            }
          }
          enableFilter={false}
          headerHeight={44}
          id="system-message-table"
          infiniteInitialRowCount={5}
          maxBlocksInCache={2}
          onBodyScroll={[Function]}
          onCellClicked={[Function]}
          onCellContextMenu={[Function]}
          onGridReady={[Function]}
          onSortChanged={[Function]}
          overlayNoRowsTemplate="No System Messages"
          pagination={false}
          rowBuffer={2}
          rowData={
            Array [
              Object {
                "category": "SOH",
                "id": "mock1",
                "message": "mock message text",
                "severity": "CRITICAL",
                "subCategory": "STATION",
                "time": 1593030714074,
                "type": "STATION_NEEDS_ATTENTION",
              },
              Object {
                "category": "SOH",
                "id": "mock2",
                "message": "mock message text",
                "severity": "CRITICAL",
                "subCategory": "STATION",
                "time": 1593030714075,
                "type": "STATION_NEEDS_ATTENTION",
              },
            ]
          }
          rowHeight={40}
          rowModelType="infinite"
          suppressCellSelection={true}
          suppressScrollOnNewData={true}
        >
          <AgGridReact
            alwaysShowVerticalScroll={true}
            animateRows={false}
            blockLoadDebounceMillis={60}
            cacheBlockSize={100}
            cacheOverflowSize={10}
            columnDefs={
              Array [
                Object {
                  "colId": "id",
                  "field": "id",
                  "headerName": "ID",
                  "headerTooltip": "ID",
                  "hide": true,
                  "width": 10,
                },
                Object {
                  "cellRenderer": undefined,
                  "cellRendererFramework": [Function],
                  "colId": "time",
                  "headerComponent": undefined,
                  "headerComponentFramework": undefined,
                  "headerName": "Timestamp",
                  "headerTooltip": "Timestamp",
                  "sort": "asc",
                  "valueFormatter": [Function],
                  "valueGetter": [Function],
                  "width": 200,
                },
                Object {
                  "cellRenderer": undefined,
                  "cellRendererFramework": [Function],
                  "colId": "category",
                  "comparator": [Function],
                  "headerComponent": undefined,
                  "headerComponentFramework": undefined,
                  "headerName": "Category",
                  "headerTooltip": "Category",
                  "valueGetter": [Function],
                  "width": 110,
                },
                Object {
                  "cellRenderer": undefined,
                  "cellRendererFramework": [Function],
                  "colId": "subCategory",
                  "comparator": [Function],
                  "headerComponent": undefined,
                  "headerComponentFramework": undefined,
                  "headerName": "Subcategory",
                  "headerTooltip": "Subcategory",
                  "valueGetter": [Function],
                  "width": 150,
                },
                Object {
                  "cellRenderer": undefined,
                  "cellRendererFramework": [Function],
                  "colId": "severity",
                  "comparator": [Function],
                  "headerComponent": undefined,
                  "headerComponentFramework": undefined,
                  "headerName": "Severity",
                  "headerTooltip": "Severity",
                  "valueGetter": [Function],
                  "width": 105,
                },
                Object {
                  "cellRenderer": undefined,
                  "cellRendererFramework": [Function],
                  "colId": "message",
                  "comparator": [Function],
                  "flex": 1,
                  "headerComponent": undefined,
                  "headerComponentFramework": undefined,
                  "headerName": "Message",
                  "headerTooltip": "Message",
                  "minWidth": 200,
                  "valueGetter": [Function],
                },
              ]
            }
            context={
              Object {
                "columnApi": undefined,
                "tableApi": undefined,
              }
            }
            datasource={null}
            debug={false}
            defaultColDef={
              Object {
                "cellRendererFramework": [Function],
                "filter": false,
                "headerComponentParams": Object {
                  "template": "
  <div class=\\"ag-cell-label-container\\" role=\\"presentation\\">
      <span ref=\\"eMenu\\" class=\\"ag-header-icon ag-header-cell-menu-button\\"></span>
      <div ref=\\"eLabel\\" class=\\"ag-header-cell-label\\" role=\\"presentation\\">
          <span ref=\\"eText\\" class=\\"ag-header-cell-text\\" role=\\"columnheader\\"></span>
          <span ref=\\"eFilter\\" class=\\"ag-header-icon ag-filter-icon\\"></span>
          <!-- <span ref=\\"eSortOrder\\" class=\\"ag-header-icon ag-sort-order\\"></span> -->
          <span ref=\\"eSortAsc\\" class=\\"ag-header-icon ag-sort-ascending-icon\\"></span>
          <span ref=\\"eSortDesc\\" class=\\"ag-header-icon ag-sort-descending-icon\\"></span>
          <span ref=\\"eSortNone\\" class=\\"ag-header-icon ag-sort-none-icon\\"></span>
      </div>
  </div>
",
                },
                "lockVisible": true,
                "resizable": true,
                "sortable": true,
                "sortingOrder": Array [
                  "asc",
                  "desc",
                ],
                "suppressCellFlash": true,
              }
            }
            deltaRowDataMode={true}
            enableFilter={false}
            getRowNodeId={[Function]}
            headerHeight={44}
            id="system-message-table"
            infiniteInitialRowCount={5}
            maxBlocksInCache={2}
            onBodyScroll={[Function]}
            onCellClicked={[Function]}
            onCellContextMenu={[Function]}
            onColumnMoved={[Function]}
            onGridReady={[Function]}
            onSortChanged={[Function]}
            overlayNoRowsTemplate="No System Messages"
            pagination={false}
            preventDefaultOnContextMenu={true}
            rowBuffer={2}
            rowData={null}
            rowDataChangeDetectionStrategy="IdentityCheck"
            rowHeight={40}
            rowModelType="infinite"
            suppressCellSelection={true}
            suppressContextMenu={true}
            suppressLoadingOverlay={true}
            suppressScrollOnNewData={true}
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </AgGridReact>
        </CoreTable>
      </InfiniteTable>
    </Component>
    <Component
      handleNewMessageIndicatorClick={[Function]}
      isVisible={false}
    />
  </div>
</SystemMessageTable>
`;
